### PR TITLE
Sign Up Backend

### PR DESCRIPTION
**Pull request by Dinuuu**
### Summary
- New users can now be created
- Change Retrofit Version

### Trello Card
https://trello.com/c/eY0dYDTN/16-tech-signup

### CODE REVIEW

File: WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/presenter/SignUpPresenter.java [hunk 1]
```diff
@@ -11,8 +23,34 @@ public SignUpPresenter(SignUpView viewInstance, Context context) {
...
+                    @Override
+                    public void onCallFailed(ResponseBody responseBody, int code) {
+                        Converter<ResponseBody, ErrorResponse> errorConverter =
+                                RetrofitServices.responseConverter(ErrorResponse.class);
+                        ErrorResponse error = null;
+                        try {
+                            error = errorConverter.convert(responseBody);
+                            getView().reportError(error.getCode());
+                        } catch (IOException e) {
+                            Log.d("SignUpPresenter",
```
Comment (framundo) [inline WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/presenter/SignUpPresenter.java]: Remove this log

File: WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/presenter/SignUpPresenter.java [hunk 2]
```diff
@@ -11,8 +23,34 @@ public SignUpPresenter(SignUpView viewInstance, Context context) {
...
+                                    "Trying to convert responseBody to ResponseError buy failed");
+                        }
+                    }
 
-        return;
+                    @Override
+                    public void onCallFailure(Throwable t) {
+                        getView().reportFailure();
```
Comment (framundo) [inline WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/presenter/SignUpPresenter.java]: The logic to interpret different error codes should be here and not in the view (fragment). The view can expose different methods "error()" and "signupError()" or whatever

File: WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/presenter/SignUpPresenter.java (hunk 1 above)
Comment (Dinuuu) [inline WoloxAndroidBootstrap/app/src/main/java/ar/com/wolox/android/presenter/SignUpPresenter.java]: What should I do with the exception ?

**Comment by framundo** [comment]
👍 Merge this

**Pull request merged**